### PR TITLE
Tests! Woo!

### PR DESCRIPTION
**Pull request by blongden**

### CODE REVIEW

**Comment by blongden** [comment]
It's only really testing the setup build directory function at the moment, but there's a fair bit of refactoring to make it more modular / easy to test - so would like to get what's done so far merged to prevent merge conflicts later.

File: libs/holograph.js [hunk 1]
```diff
@@ -117,7 +84,7 @@ function prepareCategories(results, config) {
             if (!(pages.hasOwnProperty(content.meta.category))) {
                 pages[content.meta.category] = [];
             }
-            //content.html += extractPalette(file, config);
+            content.html += extractPalette(file, config);
```
Comment (chicgeek) [inline libs/holograph.js]: ಠ_ಠ

Comment (blongden) [inline libs/holograph.js]: SORRY

File: test/err.js [hunk 2]
```diff
@@ -1,27 +1,86 @@
 var expect = require("chai").expect;
```
Comment (chicgeek) [inline test/err.js]: Can we please use single quotes throughout? We should probably use a linter for this eventually. Consistency. :)

Something to clean up later.

**Comment by blongden** [comment]
❤️

**Pull request merged**